spec/defines/filter/cors_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper'
+describe 'repose::filter::cors', :type => :define do
+  let :pre_condition do
+    'include repose'
+  end
+  context 'on RedHat' do
+    let :facts do
+    {
+      :osfamily               => 'RedHat',
+      :operationsystemrelease => '6',
+    }
+    end
+
+
+    context 'default parameters' do
+      let(:title) { 'default' }
+      it {
+        should raise_error(Puppet::Error,/nodes is a required parameter/)
+      }
+    end
+
+    context 'with ensure absent' do
+      let(:title) { 'default' }
+      let(:params) { {
+        :ensure => 'absent'
+      } }
+      it {
+        should contain_file('/etc/repose/cors.cfg.xml').with_ensure(
+          'absent')
+      }
+    end
+
+    context 'providing a validator' do
+      let(:title) { 'validator' }
+      let(:params) { {
+        :ensure     => 'present',
+        :filename   => 'cors.cfg.xml',
+        :allowed_origins      => { 10: {is_regex: 'true', origin: '.*' },
```

Comment:
it doesn't seem to like your nested chasing here.  I think the is_regex and origin not being symbols or strings anywhere in this might be the issue